src/app/EventManagement.cpp

```diff
@@ -748,6 +751,71 @@ CHIP_ERROR EventManagement::FetchEventsSince(TLVWriter & aWriter, const ObjectLi
     return err;
 }
 
+CHIP_ERROR EventManagement::RemoveInvalidFabricCB(const TLV::TLVReader & aReader, size_t aDepth, void * apContext)
+{
+    // the function does not actually remove the event, instead, it sets the fabric index to an invalid value.
+    FabricIndex * invalidFabricIndex = static_cast<FabricIndex *>(apContext);
+
+    TLVReader event;
+    TLVType tlvType;
+    TLVType tlvType1;
+    event.Init(aReader);
+    ReturnErrorOnFailure(event.EnterContainer(tlvType));
+    ReturnErrorOnFailure(event.Next());
+    ReturnErrorOnFailure(event.EnterContainer(tlvType1));
+    ReturnErrorOnFailure(event.Next());
+
+    uint8_t fabricIndex = 0;
+    while (CHIP_NO_ERROR == event.Next())
```

Comment:
So this is a _second_ `Next` on `event` inside this container, right?  What happens if the fabric index was the first thing encoded?  I know generally it's not, but why are we even making this assumption?

Reply:
agree. just remove the next in line 766, thanks